Development workflow prerequisites (#66)
infinum/default_rails_template: docs/development_workflow.md

# Prerequisites
Before developer starts working on a project he needs to get access to the
* project in the project management software (productive, jira, etc)
* project Slack channels

> * project Google Drive

Reviewer: projects 1password vault

Reviewer 2: @minazek please add 1password vault as well. 

Also, I would add comments here as well so everyone knows what needs to be filled out.

* git repository 
* development and staging secrets
* bugsnag project
* semaphore project

> * staging server

Reviewer 2: I would maybe separate this into 2 sections:

Before developer starts working on a project he needs to get access to project related stuff.
PM can give you access to:
* project in the project management software (productive, jira, etc) 
* project Slack channels
* project Google Drive

DevOps can give you access to:
* git repository 
* development and staging secrets
* bugsnag project
* semaphore project
* staging server

Additionally, I would populate this with examples of data that is expected to be filled out and who needs to fill it out, similar how we have in techical documentation:

* domain: **APP-DOMAIN** <!-- cekila.byinfinum.co --> <!-- DEVOPS -->
* user: **DEPLOY-USER** <!-- cekila_deploy --> <!-- DEVOPS -->
* ruby version: **RUBY VERSION** <!-- 2.7.1 --> <!-- DEVELOPER -->
* node version: **NODE VERSION** <!-- 14.0.1 --> <!-- DEVELOPER -->

E.g. 

* bugsnag project - **LINK-TO-BUGSNAG** <!-- https://bugsnag.com/path-to-project --> <!-- DEVELOPER -->

So the info would be:
* git repository - link to repo
* development and staging secrets - link to repo
* bugsnag project - link to bugsnag
* semaphore project - link to semaphore
* staging server - link to staging

Also add a note for devs to add all other project specific stuff here that a developer needs access to. We've listed only defaults for all projects.

Reviewer: The question is:
should this go through #devops-hotline or through task?
either way we might create a "template" which can be pasted into slack/task

Reviewer 2: > Dont know if we need the 3rd party things as we should already have those in https://github.com/infinum/default_rails_template/blob/master/docs/architecture/services.md

Hmm, i was thinking about it, but then, we sometimes have a lot of external services that you don't need access to right away or you'll ever need to (if you're just temporarily working on a project). 

With this list I wanted to cover things that are prerequisites for basic development. 

> should this go through #devops-hotline or through task?

Initially we were thinking of tasks, but i think you usually need access to a project asap so we usually do it through devops-hotline. Either way, the idea is to paste the list for devops related items and ask for access.

Reviewer: > Hmm, i was thinking about it, but then, we sometimes have a lot of external services that you don't need access to right away or you'll ever need to (if you're just temporarily working on a project).

Ok

> Initially we were thinking of tasks, but i think you usually need access to a project asap so we usually do it through devops-hotline. Either way, the idea is to paste the list for devops related items and ask for access.

how about we open a task and in #devops-channel just link to the task? :D

Reviewer 2: > how about we open a task and in #devops-channel just link to the task? :D

where would you open this task?
- project-project on productive - in most cases a devops won't have access to that project
- devops project on productive - not sure if all developers have access to that project

Reviewer: Good points

Would actually ask devops what would they prefer.
I know that at least one devops is on each project because they needed to setup the server :D

Reviewer 2: Yeah, i know, but the point of devops-hotline is that you get help from any devops that is free so you don't have to wait for a devops that was involved in setting up the project. 

I'll check with Furlan.

# Prerequisites

> Before developer starts working on a project he needs to get access project related items.

Reviewer 3: ```suggestion
Before you start working on the project, you need to get access to project-related items.
```

Project manager can give you access to:
* project in the project management software (productive, jira, etc) **LINK-TO-PROJECT** <!-- https://app.productive.io/path-to-project --> <!-- DEVELOPER -->
* project Slack channels **LINK-TO-SLACK-CHANNELS** <!-- https://infinum.slack.com/path-to-project --> <!-- DEVELOPER -->
* project Google Drive **LINK-TO-GOOGLE-DRIVE** <!-- https://drive.google.com//path-to-project --> <!-- DEVELOPER -->
* project design (Figma, Sketch, etc) **LINK-TO-DESIGN** <!-- https://www.figma.com/path-to-project --> <!-- DEVELOPER -->
  
Devops can give you access to:

> * 1password vault

Reviewer 2: Maybe just add - `**VAULT-NAME**`
